source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRImageRenderer.mm

```diff
@@ -122,16 +122,24 @@ - (UIView* )render:(UIView* )viewGroup
      andHostConfig:(std::shared_ptr<HostConfig> const &)config
 {
     std::shared_ptr<Image> imgElem = std::dynamic_pointer_cast<Image>(elem);
+    NSString* urlStr = [NSString stringWithCString: imgElem->GetUrl().c_str()
+                                          encoding:[NSString defaultCStringEncoding]];
+    NSURL* url = nil;
+    NSRange range = [urlStr rangeOfString:@"%"];
+    //check if % is already embedded in urlStr
+    if(range.length > 0)
+    {
+        url = [NSURL URLWithString:urlStr];
+    }
+    else
+    {
+        url = [NSURL URLWithString: 
+                [urlStr stringByAddingPercentEncodingWithAllowedCharacters:NSCharacterSet.URLFragmentAllowedCharacterSet]];
+    }
```

Comment:
I don't like this check. The GetUrl method should be responsible for deciding whether the value it's returning is percent-encoded or not (probably yes, if it's purporting to be a valid URL), rather than leaving it up to the caller to try to guess with heuristics.